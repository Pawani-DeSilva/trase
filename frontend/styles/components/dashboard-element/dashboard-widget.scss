.c-dashboard-widget {
  padding: 40px 0 0;

  .widget-title-container {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
  }

  .widget-title {
    color: $white;
    font-family: $font-family-2;
    font-size: 20px;
    font-weight: $font-weight-light;
    line-height: 1.6;
    letter-spacing: -0.4px;
  }

  .widget-actions {
    button {
      cursor: pointer;
      width: 30px;
      height: 30px;
      border-radius: 100%;
      border: solid 1px rgba($white, 0.13);
      margin: 0 15px 0 0;

      &:last-child {
        margin: 0;
      }
    }
  }

  .widget-box {
    height: 480px;
    border-radius: 6px;
    border: solid 1px rgba($white, 0.13);
    margin: 18px 0 0;

    .widget-spinner {
      display: flex;
      align-items: center;
      justify-content: center;
      height: 100%;
      width: 100%;
    }
  }
}
