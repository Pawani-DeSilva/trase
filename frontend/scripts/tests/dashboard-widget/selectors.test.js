import { makeGetConfig } from 'react-components/dashboard-element/dashboard-widget/dashboard-widget.selectors';
import CHART_CONFIG from 'react-components/dashboard-element/dashboard-widget/dashboard-widget-config';

const meta = {
  xAxis: {
    label: 'Year',
    prefix: '',
    format: '',
    suffix: ''
  },
  yAxis: {
    label: 'Water scarcity',
    prefix: '',
    format: '',
    suffix: '/7'
  },
  x: {
    type: 'category',
    label: 'Year',
    tooltip: {
      prefix: '',
      format: '',
      suffix: ''
    }
  },
  y1: {
    type: 'number',
    label: '',
    tooltip: {
      prefix: '',
      format: '',
      suffix: ''
    }
  }
};

// makeGetConfig is a function that returns the selector just to be able to memoize the selector result
describe('Widget config parse selectors', () => {
  describe('makeGetConfig', () => {
    it('returns the defaultConfig if there is no meta', () => {
      expect(makeGetConfig()(null, { meta: null })).toEqual(CHART_CONFIG.line);
    });

    it('Parses the config', () => {
      expect(makeGetConfig()(null, { meta, chartType: 'line' })).toMatchSnapshot();
    });
  });
});
