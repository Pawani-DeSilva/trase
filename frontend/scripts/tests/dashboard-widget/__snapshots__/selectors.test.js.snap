// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget config parse selectors makeGetConfig Parses the config 1`] = `
Object {
  "cartesianGrid": Object {
    "strokeDasharray": "5 5",
    "vertical": false,
  },
  "colors": Array [
    Object {
      "color": "#fff0c2",
      "label": "Water scarcity",
    },
  ],
  "tooltip": Object {
    "content": null,
    "cursor": Object {
      "opacity": 0.5,
      "stroke": "#9a1e2a",
      "strokeWidth": 2,
    },
  },
  "type": "line",
  "xAxis": Object {
    "axisLine": false,
    "interval": "preserveStartEnd",
    "tick": Object {
      "fill": "#ffffff",
      "fontFamily": "DecimaMonoPro",
      "fontSize": 14,
    },
    "tickLine": false,
    "type": "category",
  },
  "xKey": "x",
  "yAxis": Object {
    "axisLine": false,
    "domain": Array [
      0,
      "auto",
    ],
    "tick": Object {
      "fill": "#ffffff",
      "fontFamily": "DecimaMonoPro",
      "fontSize": 14,
      "interval": 2,
      "tickCount": 2,
    },
    "tickFormatter": [Function],
    "type": undefined,
  },
  "yAxisLabel": Object {
    "suffix": "/7",
    "text": "Water scarcity",
  },
  "yKeys": Object {
    "lines": Object {
      "y1": Object {
        "fill": "#fff0c2",
        "stroke": "#fff0c2",
      },
    },
  },
}
`;
